posthog/api/file_system.py

```diff
@@ -138,12 +139,18 @@ def safely_get_queryset(self, queryset: QuerySet) -> QuerySet:
         if self.action == "list":
             queryset = queryset.order_by("path")
 
+        if self.user_access_control:
+            queryset = self.user_access_control.filter_and_annotate_file_system_queryset(queryset)
+
         return queryset
 
     def destroy(self, request, *args, **kwargs):
         instance = self.get_object()
         if instance.type == "folder":
-            FileSystem.objects.filter(team=self.team, path__startswith=f"{instance.path}/").delete()
+            qs = FileSystem.objects.filter(team=self.team, path__startswith=f"{instance.path}/")
+            if self.user_access_control:
+                qs = self.user_access_control.filter_and_annotate_file_system_queryset(qs)
```

Comment:
Interesting - what happens to things in it where they don't have access? This is just deleting the file system not the actual resource, right?